src/mongo/s/stats/joom_latency_histogram.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include <array>
+
+#include "mongo/db/commands.h"
+
+namespace mongo {
+
+class BSONObjBuilder;
+
+enum class CommandName {
+    Insert,
+    Find,
+    FindAndModify,
+    Update,
+    Delete,
+    Aggregate,
+    Distinct,
+    Other,
+};
+
+class JoomLatencyHistogram {
+public:
+    static const int kMaxBuckets = 10;
+    static const std::array<uint64_t, kMaxBuckets> kUpperBoundsMicros;
+    void increment(CommandName cmdName, uint64_t latency, bool isError);
+    void append(BSONObjBuilder* builder) const;
+
+private:
+    struct HistogramData {
+        std::array<uint64_t, kMaxBuckets> buckets{};
+        uint64_t sum = 0;
+        uint64_t successCount = 0;
+        uint64_t errorCount = 0;
```

Comment:
In my opinion, it's better to stick with a single style for default value - `uint64_t errorCount{}`